// User defined proto file to use Ray Serve gRPC proxy
// generate python code:
// cd <to the upper directory of this file>
// python -m grpc_tools.protoc -I./protos --python_out=. --pyi_out=. --grpc_python_out=. ./protos/user_defined_protos.proto

syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.ray.examples.user_defined_protos";
option java_outer_classname = "UserDefinedProtos";

package userdefinedprotos;

message UserDefinedMessage {
  string name = 1;
  string foo = 2;
  int64 num = 3;
}

message UserDefinedResponse {
  string greeting = 1;
  int64 num_x2 = 2;
}

message UserDefinedMessage2 {}

message UserDefinedResponse2 {
  string greeting = 1;
}

message FruitAmounts {
    int64 orange = 1;
    int64 apple = 2;
    int64 banana = 3;
}

message FruitCosts {
    float costs = 1;
}

service UserDefinedService {
  rpc __call__(UserDefinedMessage) returns (UserDefinedResponse);
  rpc Method1(UserDefinedMessage) returns (UserDefinedResponse);
  rpc Method2(UserDefinedMessage2) returns (UserDefinedResponse2);
  rpc Streaming(UserDefinedMessage) returns (stream UserDefinedResponse);
}

service FruitService {
  rpc FruitStand(FruitAmounts) returns (FruitCosts);
}

message RoutesRequest {}

message RoutesResponse {
  repeated string application_names = 1;
}

message HealthzRequest {}

message HealthzResponse {
  string response = 1;
}

service ServeAPIService {
  rpc ServeRoutes(RoutesRequest) returns (RoutesResponse);
  rpc ServeHealthz(HealthzRequest) returns (HealthzResponse);
}
